pages/_app/index.jsx

```diff
@@ -32,23 +20,48 @@ class App extends NextApp {
         (await Component.getInitialProps({ ...ctx, initialStoreData })) || {}
     }
 
+    const route = new Route(
+      (ctx.req && ctx.req.url) || window.location.pathname
```

Comment:
when `ctx.req` is empty?

Reply:
Oh, I do not use it anymore! Thank you!

It's empty on client side and fulfilled on server side.